editor-components/modal/confirmation-modal-1/confirmation.tsx

```diff
@@ -0,0 +1,117 @@
+            value: "Hang On a Sec!"
+        })
+
+        this.addProp({
+            type: "string",
+            key: "description",
+            displayer: "Description",
+            value: "Ready to rock and roll? Confirm your choice, and let's make some magic happen!"
+        })
+
+        this.addProp({
+            type: "object",
+            key: "button-1",
+            displayer: "Button",
+            value: [
+                {
+                    type: "string",
+                    key: "button1-text",
+                    displayer: "Button Text",
+                    value: "Count Me In"
+                },
+                {
+                    type: "page",
+                    key: "button-path",
+                    displayer: "Button Path",
+                    value: "// TODO"
+                }
+            ]
+        })
+        
+        this.addProp({
+            type: "object",
+            key: "button-2",
+            displayer: "Button 2",
+            value: [
+                {
+                    type: "string",
+                    key: "button2-text",
+                    displayer: "Button Text",
+                    value: "Let Me Rethink"
+                }
+            ]
+        })
+    }
+
+
+
+    getName(): string {
+        return "Confirmation Modal";
+    }
+    
+    render() {
+        
+        return (
+            <div className={this.decorateCSS("container")}>
+                <div className={this.decorateCSS("max-content")}>
+                    <div className={this.decorateCSS("box")}>                 
+                        <div className={this.decorateCSS("head-content")}>
+                            <img 
+                                alt= ""
+                                className={this.decorateCSS("image")}
+                                src={image1}                        
+                            ></img>
+                            <ComposerModalClose>
+                                <button className={this.decorateCSS("close-button")}>&#10005;</button>
+                            </ComposerModalClose>
+                        </div>
+                        <div className={this.decorateCSS("bottom-content")}>
+                            <h1 className={this.decorateCSS("title")}>
+                                {this.getPropValue("title")}
+                            </h1>
+                            <p className={this.decorateCSS("description")}>
+                                {this.getPropValue("description")}
+                            </p>
+
+                            <ComposerModalClose>
+                            <div className={this.decorateCSS("buttons")}>
+                                <ComposerLink path={this.getPropValue("button-1")[1].value}>
+                                    <button className={this.decorateCSS("button-1")} >
+                                        {this.getPropValue("button-1")[0].value}
```

Comment:
do not access values like this. create a type.
also do not render button if button text is empty.